File: Year 2/Data Structures & Algorithms/week01/SearchArray.java
/*
  Write a Java program to caluculate the time taken to search for an element in
  an array
  
  @author Shrey Mathur
 */

import java.util.Scanner;

class SearchArray {

	private static boolean check(int a[], int n, int e) {
		boolean isFound = false;
		for (int i = 0; i < n; i++) {
			if (a[i] == e) {
				isFound = true;
				break;
			}
		}
		return isFound;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter number of elemnts");
		int n = sc.nextInt();

		int ar[] = new int[n];
		System.out.println("Enter " + n + " values for the array");
		for (int i = 0; i < n; i++)
			ar[i] = sc.nextInt();

		System.out.println("Enter number to be searched");
		int e = sc.nextInt();

		long startTime = System.currentTimeMillis();
		boolean isFound = check(ar, n, e);
		if (isFound)
			System.out.println("Element found");
		else
			System.out.println("Element is not found");
		long endTime = System.currentTimeMillis();
		System.out.println("The start time is " + startTime);
		System.out.println("The end time is " + endTime);
		long totalTime = endTime - startTime;
		System.out.println("The total time taken is " + totalTime);
		sc.close();
	}
}

// OUTPUT
// Enter number of elemnts
// 3
// Enter 3 values for the array
// 1 2 3
// Enter
// number to be searched
// 2
// Element found
// The start time is 1617256038871
// The end time is 1617256038871
// The total time taken is 0

File: Year 2/Data Structures & Algorithms/week01/SearchArray.c
// Write a C program to check
#include <stdio.h>
#include <time.h>

void check(int a[], int n, int e) {
	int i, flag = 0;
	for (i = 0;i < n;i++) {
		if (a[i] == e) {
			flag = 1;
			break;
		}
	}
	if (flag == 0) printf("Element was not found\n");
	else printf("element was found\n");
}
void main() {
	clock_t start_t, end_t, total_t;
	int n, e;
	printf("Enter the vale of n:\n");
	scanf("%d", &n);
	int a[n];
	printf("Enter %d elements for the array:\n", n);
	for (int i = 0; i < n; ++i) scanf("%d", &a[i]);

	printf("\nEnter element to be searched\n");
	scanf("%d", &e);
	start_t = clock();
	check(a, n, e);
	end_t = clock();
	printf("Start_t = %ld\n", start_t);
	printf("End time = %ld\n", end_t);
	total_t = (double)(end_t - start_t) / CLOCKS_PER_SEC;
	printf("Total time taken by CPU: %f\n", total_t);
}
